src/core.ts

```diff
@@ -119,12 +119,23 @@ export function bind(target: EventTarget, event: string, listener: EventListener
 }
 
 export function addEvent(event: IEventData, scheduleUpload: boolean = true): void {
-  let evtJson: IEvent = {
-    id: eventCount++,
-    time: isNumber(event.time) ? event.time : getTimestamp(),
-    type: event.type,
-    state: event.state
-  };
+  const stateLength = event.state ? JSON.stringify(event.state).length : -1;
+  let evtJson: IEvent = config.eventLimit && stateLength > config.eventLimit
+    ? {
+      id: eventCount++,
+      time: isNumber(event.time) ? event.time : getTimestamp(),
+      type: InstrumentationEventName,
+      state: {
+        type: Instrumentation.OversizedEvent,
+        size: stateLength
```

Comment:
>stateLength [](start = 14, length = 11)

Include more information about the event. If you can send the event "shell" (the event types, etc.